hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -230,6 +233,9 @@ public Response put(
         throw os3Exception;
       } else if (ex.getResult() == ResultCodes.PERMISSION_DENIED) {
         throw newError(S3ErrorTable.ACCESS_DENIED, keyPath, ex);
+      } else if (ex.getResult() == ResultCodes.BUCKET_NOT_FOUND
+          || ex.getResult() == ResultCodes.VOLUME_NOT_FOUND) {
+        throw newError(S3ErrorTable.NO_SUCH_BUCKET, bucketName, ex);
```

Comment:
Thanks for the detailed review @smengcl. 
We already log the original exception in the audit log a few lines before. And this part of the code is only to help translate OMException to S3 API responses where the volume concept is not a part of.

Reply:
Got it.

Does `NO_SUCH_VOLUME` work for `VOLUME_NOT_FOUND` here?

Reply:
Thanks @smengcl for the review. Volume not found should actually be treated differently than bucket not found. Volume not found would indicate a misconfigured system and should show up in both audit and s3g log. Bucket not found should only be in audit log as this is most likely a misconfigured client.

Reply:
I'm not sure why there's a `NO_SUCH_VOLUME` in S3ErrorTable and it's unused now. If we use it, S3 client would yield an error message like:
```
An error occurred (NoSuchVolume) when calling the operation: The specified volume does not exist.
```
And that sounds a bit weird for the S3 clients as they're not supposed to create a volume.

Reply:
got it, thanks @kerneltime @smengcl  for identifying this. I've update the code so that VOLUME_NOT_FOUND is treated as a system error (client receives `INTERNAL_SERVER_ERROR` and errors is in both logs).

Reply:
Great! Thanks @duongnguyen0 .